### ISSUE TITLE
Always get errno 32 if frame brackets are set to 32

### ISSUE THREAD

**Issue by Basedbase-ai** [tdrussell/diffusion-pipe#161]
EDIT: Heres the console log It happens anytime I try to train above 30 frames locally despite having block_swap set to 40 and 128GB of system ram resolution is set to train at [512] always crashes
  loss: loss_fn
[2025-03-16 19:15:33,998] [INFO] [logging.py:128:log_dist] [Rank 0] DeepSpeed info: version=0.16.4, git-hash=unknown, git-branch=unknown
[2025-03-16 19:15:33,999] [INFO] [config.py:734:__init__] Config mesh_device None world_size = 1
[2025-03-16 19:15:34,220] [INFO] [logging.py:128:log_dist] [Rank 0] DeepSpeed Flops Profiler Enabled: False
[2025-03-16 19:15:34,266] [INFO] [logging.py:128:log_dist] [Rank 0] Using client callable to create basic optimizer
[2025-03-16 19:15:34,266] [INFO] [logging.py:128:log_dist] [Rank 0] Removing param_group that has no 'params' in the basic Optimizer
[2025-03-16 19:15:34,349] [INFO] [logging.py:128:log_dist] [Rank 0] DeepSpeed Basic Optimizer = AdamW
[2025-03-16 19:15:34,349] [INFO] [logging.py:128:log_dist] [Rank 0] DeepSpeed Final Optimizer = AdamW
[2025-03-16 19:15:34,349] [INFO] [logging.py:128:log_dist] [Rank 0] DeepSpeed using configured LR scheduler = None
[2025-03-16 19:15:34,349] [INFO] [logging.py:128:log_dist] [Rank 0] DeepSpeed LR Scheduler = None
[2025-03-16 19:15:34,350] [INFO] [logging.py:128:log_dist] [Rank 0] step=0, skipped=0, lr=[0.0001], mom=[0.0]
[2025-03-16 19:15:34,356] [INFO] [config.py:1001:print] DeepSpeedEngine configuration:
[2025-03-16 19:15:34,356] [INFO] [config.py:1005:print]   activation_checkpointing_config  {
    "partition_activations": false,
    "contiguous_memory_optimization": false,
    "cpu_checkpointing": false,
    "number_checkpoints": null,
    "synchronize_checkpoint_boundary": false,
    "profile": false
}
[2025-03-16 19:15:34,356] [INFO] [config.py:1005:print]   aio_config ................... {'block_size': 1048576, 'queue_depth': 8, 'intra_op_parallelism': 1, 'single_submit': False, 'overlap_events': True, 'use_gds': False}
[2025-03-16 19:15:34,356] [INFO] [config.py:1005:print]   amp_enabled .................. False
[2025-03-16 19:15:34,356] [INFO] [config.py:1005:print]   amp_params ................... False
[2025-03-16 19:15:34,356] [INFO] [config.py:1005:print]   autotuning_config ............ {
    "enabled": false,
    "start_step": null,
    "end_step": null,
    "metric_path": null,
    "arg_mappings": null,
    "metric": "throughput",
    "model_info": null,
    "results_dir": "autotuning_results",
    "exps_dir": "autotuning_exps",
    "overwrite": true,
    "fast": true,
    "start_profile_step": 3,
    "end_profile_step": 5,
    "tuner_type": "gridsearch",
    "tuner_early_stopping": 5,
    "tuner_num_trials": 50,
    "model_info_path": null,
    "mp_size": 1,
    "max_train_batch_size": null,
    "min_train_batch_size": 1,
    "max_train_micro_batch_size_per_gpu": 1.024000e+03,
    "min_train_micro_batch_size_per_gpu": 1,
    "num_tuning_micro_batch_sizes": 3
}
[2025-03-16 19:15:34,357] [INFO] [config.py:1005:print]   bfloat16_enabled ............. False
[2025-03-16 19:15:34,357] [INFO] [config.py:1005:print]   bfloat16_immediate_grad_update  False
[2025-03-16 19:15:34,357] [INFO] [config.py:1005:print]   checkpoint_parallel_write_pipeline  False
[2025-03-16 19:15:34,357] [INFO] [config.py:1005:print]   checkpoint_tag_validation_enabled  True
[2025-03-16 19:15:34,357] [INFO] [config.py:1005:print]   checkpoint_tag_validation_fail  False
[2025-03-16 19:15:34,357] [INFO] [config.py:1005:print]   comms_config ................. <deepspeed.comm.config.DeepSpeedCommsConfig object at 0x7fea37555e50>
[2025-03-16 19:15:34,357] [INFO] [config.py:1005:print]   communication_data_type ...... None
[2025-03-16 19:15:34,357] [INFO] [config.py:1005:print]   compression_config ........... {'weight_quantization': {'shared_parameters': {'enabled': False, 'quantizer_kernel': False, 'schedule_offset': 0, 'quantize_groups': 1, 'quantize_verbose': False, 'quantization_type': 'symmetric', 'quantize_weight_in_forward': False, 'rounding': 'nearest', 'fp16_mixed_quantize': False, 'quantize_change_ratio': 0.001}, 'different_groups': {}}, 'activation_quantization': {'shared_parameters': {'enabled': False, 'quantization_type': 'symmetric', 'range_calibration': 'dynamic', 'schedule_offset': 1000}, 'different_groups': {}}, 'sparse_pruning': {'shared_parameters': {'enabled': False, 'method': 'l1', 'schedule_offset': 1000}, 'different_groups': {}}, 'row_pruning': {'shared_parameters': {'enabled': False, 'method': 'l1', 'schedule_offset': 1000}, 'different_groups': {}}, 'head_pruning': {'shared_parameters': {'enabled': False, 'method': 'topk', 'schedule_offset': 1000}, 'different_groups': {}}, 'channel_pruning': {'shared_parameters': {'enabled': False, 'method': 'l1', 'schedule_offset': 1000}, 'different_groups': {}}, 'layer_reduction': {'enabled': False}}
[2025-03-16 19:15:34,357] [INFO] [config.py:1005:print]   curriculum_enabled_legacy .... False
[2025-03-16 19:15:34,357] [INFO] [config.py:1005:print]   curriculum_params_legacy ..... False
[2025-03-16 19:15:34,357] [INFO] [config.py:1005:print]   data_efficiency_config ....... {'enabled': False, 'seed': 1234, 'data_sampling': {'enabled': False, 'num_epochs': 1000, 'num_workers': 0, 'curriculum_learning': {'enabled': False}}, 'data_routing': {'enabled': False, 'random_ltd': {'enabled': False, 'layer_token_lr_schedule': {'enabled': False}}}}
[2025-03-16 19:15:34,357] [INFO] [config.py:1005:print]   data_efficiency_enabled ...... False
[2025-03-16 19:15:34,358] [INFO] [config.py:1005:print]   dataloader_drop_last ......... False
[2025-03-16 19:15:34,358] [INFO] [config.py:1005:print]   disable_allgather ............ False
[2025-03-16 19:15:34,358] [INFO] [config.py:1005:print]   dump_state ................... False
[2025-03-16 19:15:34,358] [INFO] [config.py:1005:print]   dynamic_loss_scale_args ...... None
[2025-03-16 19:15:34,358] [INFO] [config.py:1005:print]   eigenvalue_enabled ........... False
[2025-03-16 19:15:34,358] [INFO] [config.py:1005:print]   eigenvalue_gas_boundary_resolution  1
[2025-03-16 19:15:34,358] [INFO] [config.py:1005:print]   eigenvalue_layer_name ........ bert.encoder.layer
[2025-03-16 19:15:34,358] [INFO] [config.py:1005:print]   eigenvalue_layer_num ......... 0
[2025-03-16 19:15:34,358] [INFO] [config.py:1005:print]   eigenvalue_max_iter .......... 100
[2025-03-16 19:15:34,358] [INFO] [config.py:1005:print]   eigenvalue_stability ......... 1e-06
[2025-03-16 19:15:34,358] [INFO] [config.py:1005:print]   eigenvalue_tol ............... 0.01
[2025-03-16 19:15:34,358] [INFO] [config.py:1005:print]   eigenvalue_verbose ........... False
[2025-03-16 19:15:34,358] [INFO] [config.py:1005:print]   elasticity_enabled ........... False
[2025-03-16 19:15:34,359] [INFO] [config.py:1005:print]   flops_profiler_config ........ {
    "enabled": false,
    "recompute_fwd_factor": 0.0,
    "profile_step": 1,
    "module_depth": -1,
    "top_modules": 1,
    "detailed": true,
    "output_file": null
}
[2025-03-16 19:15:34,359] [INFO] [config.py:1005:print]   fp16_auto_cast ............... None
[2025-03-16 19:15:34,359] [INFO] [config.py:1005:print]   fp16_enabled ................. False
[2025-03-16 19:15:34,359] [INFO] [config.py:1005:print]   fp16_master_weights_and_gradients  False
[2025-03-16 19:15:34,359] [INFO] [config.py:1005:print]   global_rank .................. 0
[2025-03-16 19:15:34,359] [INFO] [config.py:1005:print]   grad_accum_dtype ............. None
[2025-03-16 19:15:34,359] [INFO] [config.py:1005:print]   gradient_accumulation_steps .. 4
[2025-03-16 19:15:34,359] [INFO] [config.py:1005:print]   gradient_clipping ............ 1.0
[2025-03-16 19:15:34,359] [INFO] [config.py:1005:print]   gradient_predivide_factor .... 1.0
[2025-03-16 19:15:34,359] [INFO] [config.py:1005:print]   graph_harvesting ............. False
[2025-03-16 19:15:34,359] [INFO] [config.py:1005:print]   hybrid_engine ................ enabled=False max_out_tokens=512 inference_tp_size=1 release_inference_cache=False pin_parameters=True tp_gather_partition_size=8
[2025-03-16 19:15:34,359] [INFO] [config.py:1005:print]   initial_dynamic_scale ........ 65536
[2025-03-16 19:15:34,359] [INFO] [config.py:1005:print]   load_universal_checkpoint .... False
[2025-03-16 19:15:34,359] [INFO] [config.py:1005:print]   loss_scale ................... 0
[2025-03-16 19:15:34,359] [INFO] [config.py:1005:print]   memory_breakdown ............. False
[2025-03-16 19:15:34,359] [INFO] [config.py:1005:print]   mics_hierarchial_params_gather  False
[2025-03-16 19:15:34,360] [INFO] [config.py:1005:print]   mics_shard_size .............. -1
[2025-03-16 19:15:34,360] [INFO] [config.py:1005:print]   monitor_config ............... tensorboard=TensorBoardConfig(enabled=False, output_path='', job_name='DeepSpeedJobName') comet=CometConfig(enabled=False, samples_log_interval=100, project=None, workspace=None, api_key=None, experiment_name=None, experiment_key=None, online=None, mode=None) wandb=WandbConfig(enabled=False, group=None, team=None, project='deepspeed') csv_monitor=CSVConfig(enabled=False, output_path='', job_name='DeepSpeedJobName')
[2025-03-16 19:15:34,360] [INFO] [config.py:1005:print]   nebula_config ................ {
    "enabled": false,
    "persistent_storage_path": null,
    "persistent_time_interval": 100,
    "num_of_version_in_retention": 2,
    "enable_nebula_load": true,
    "load_path": null
}
[2025-03-16 19:15:34,360] [INFO] [config.py:1005:print]   optimizer_legacy_fusion ...... False
[2025-03-16 19:15:34,360] [INFO] [config.py:1005:print]   optimizer_name ............... None
[2025-03-16 19:15:34,360] [INFO] [config.py:1005:print]   optimizer_params ............. None
[2025-03-16 19:15:34,360] [INFO] [config.py:1005:print]   pipeline ..................... {'stages': 'auto', 'partition': 'best', 'seed_layers': False, 'activation_checkpoint_interval': 0, 'pipe_partitioned': True, 'grad_partitioned': True}
[2025-03-16 19:15:34,360] [INFO] [config.py:1005:print]   pld_enabled .................. False
[2025-03-16 19:15:34,360] [INFO] [config.py:1005:print]   pld_params ................... False
[2025-03-16 19:15:34,361] [INFO] [config.py:1005:print]   prescale_gradients ........... False
[2025-03-16 19:15:34,361] [INFO] [config.py:1005:print]   scheduler_name ............... None
[2025-03-16 19:15:34,361] [INFO] [config.py:1005:print]   scheduler_params ............. None
[2025-03-16 19:15:34,361] [INFO] [config.py:1005:print]   seq_parallel_communication_data_type  torch.float32
[2025-03-16 19:15:34,361] [INFO] [config.py:1005:print]   sparse_attention ............. None
[2025-03-16 19:15:34,361] [INFO] [config.py:1005:print]   sparse_gradients_enabled ..... False
[2025-03-16 19:15:34,361] [INFO] [config.py:1005:print]   steps_per_print .............. 1
[2025-03-16 19:15:34,361] [INFO] [config.py:1005:print]   tensor_parallel_config ....... dtype=torch.float16 autotp_size=0 tensor_parallel=TPConfig(tp_size=1, tp_grain_size=1, mpu=None, tp_group=None) injection_policy_tuple=None keep_module_on_host=False replace_with_kernel_inject=False
[2025-03-16 19:15:34,361] [INFO] [config.py:1005:print]   timers_config ................ enabled=True synchronized=True
[2025-03-16 19:15:34,361] [INFO] [config.py:1005:print]   train_batch_size ............. 4
[2025-03-16 19:15:34,361] [INFO] [config.py:1005:print]   train_micro_batch_size_per_gpu  1
[2025-03-16 19:15:34,361] [INFO] [config.py:1005:print]   use_data_before_expert_parallel_  False
[2025-03-16 19:15:34,362] [INFO] [config.py:1005:print]   use_node_local_storage ....... False
[2025-03-16 19:15:34,362] [INFO] [config.py:1005:print]   wall_clock_breakdown ......... False
[2025-03-16 19:15:34,362] [INFO] [config.py:1005:print]   weight_quantization_config ... None
[2025-03-16 19:15:34,362] [INFO] [config.py:1005:print]   world_size ................... 1
[2025-03-16 19:15:34,362] [INFO] [config.py:1005:print]   zero_allow_untested_optimizer  False
[2025-03-16 19:15:34,362] [INFO] [config.py:1005:print]   zero_config .................. stage=0 contiguous_gradients=True reduce_scatter=True reduce_bucket_size=500000000 use_multi_rank_bucket_allreduce=True allgather_partitions=True allgather_bucket_size=500000000 overlap_comm=False load_from_fp32_weights=True elastic_checkpoint=False offload_param=None offload_optimizer=None sub_group_size=1000000000 cpu_offload_param=None cpu_offload_use_pin_memory=None cpu_offload=None prefetch_bucket_size=50000000 param_persistence_threshold=100000 model_persistence_threshold=9223372036854775807 max_live_parameters=1000000000 max_reuse_distance=1000000000 gather_16bit_weights_on_model_save=False module_granularity_threshold=0 use_all_reduce_for_fetch_params=False stage3_gather_fp16_weights_on_model_save=False ignore_unused_parameters=True legacy_stage1=False round_robin_gradients=False zero_hpz_partition_size=1 zero_quantized_weights=False zero_quantized_nontrainable_weights=False zero_quantized_gradients=False zeropp_loco_param=None mics_shard_size=-1 mics_hierarchical_params_gather=False memory_efficient_linear=True pipeline_loading_checkpoint=False override_module_apply=True log_trace_cache_warnings=False
[2025-03-16 19:15:34,362] [INFO] [config.py:1005:print]   zero_enabled ................. False
[2025-03-16 19:15:34,362] [INFO] [config.py:1005:print]   zero_force_ds_cpu_optimizer .. True
[2025-03-16 19:15:34,362] [INFO] [config.py:1005:print]   zero_optimization_stage ...... 0
[2025-03-16 19:15:34,362] [INFO] [config.py:991:print_user_config]   json = {
    "train_micro_batch_size_per_gpu": 1,
    "gradient_accumulation_steps": 4,
    "gradient_clipping": 1.0,
    "steps_per_print": 1
}
[2025-03-16 19:15:34,362] [INFO] [engine.py:105:__init__] CONFIG: micro_batches=4 micro_batch_size=1
[2025-03-16 19:15:34,362] [INFO] [engine.py:146:__init__] is_pipe_partitioned= False is_grad_partitioned= False
[2025-03-16 19:15:34,424] [INFO] [engine.py:165:__init__] RANK=0 STAGE=0 LAYERS=42 [0, 42) STAGE_PARAMS=359137280 (359.137M) TOTAL_PARAMS=359137280 (359.137M) UNIQUE_PARAMS=359137280 (359.137M)
[2025-03-16 19:15:47,934] [INFO] [launch.py:319:sigkill_handler] Killing subprocess 29049
[2025-03-16 19:15:48,014] [ERROR] [launch.py:325:sigkill_handler] ['/home/corbett/miniconda3/envs/diffusion-pipe/bin/python', '-u', 'train.py', '--local_rank=0', '--deepspeed', '--config', 'examples/wan_video.toml'] exits with return code = -9
